scripts/components/link-edit-component/link-edit-component.js

```diff
@@ -0,0 +1,148 @@
+import React, { useRef } from 'react';
+import { __experimentalLinkControl as LinkControl } from '@wordpress/block-editor';
+import { Popover, Button, BaseControl } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { icons } from '@eightshift/frontend-libs/scripts/editor';
+import { truncateMiddle } from '@eightshift/frontend-libs/scripts/helpers';
+
+/**
+ * Options panel component that allows picking an URL in a clean and simple way.
+ * 
+ * @param {object} props                                        - LinkEditComponent options.
+ * @param {string?} props.url                                   - Currently selected URL.
+ * @param {boolean} props.opensInNewTab                         - Currently selected option for opening the link in a new tab.
+ * @param {function} props.setAttributes                        - The `setAttributes` callback from component/block attributes.
+ * @param {string} props.title                                  - Name of the component/block displayed in the tooltip (*Use <title>*).
+ * @param {string} props.urlAttrName                            - Name of the `url` attribute (use `getAttrKey`)
+ * @param {string} props.isNewTabAttrName                       - Name of the `isNewTab` attribute (use `getAttrKey`)
+ * @param {string?} [props.textDomain=eightshift-frontend-libs] - Text domain to use for i18n.
+ * @param {boolean} [props.showNewTabOption=true]               - If `true`, displays the 'Open in new tab' toggle.
+ */
+export const LinkEditComponent = ({
+	url,
+	opensInNewTab,
+	setAttributes,
+	urlAttrName,
+	isNewTabAttrName,
+	title,
+	textDomain = 'eightshift-frontend-libs',
+	showNewTabOption = true,
+}) => {
+	const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+	const ref = useRef();
+
+	const openLinkControl = () => {
+		setIsDropdownOpen(true);
+		return false; // Prevents default behaviour for event.
+	};
+
+	const unlinkButton = () => {
+		let newValues = {
+			[urlAttrName]: undefined,
+		};
+
+		if (showNewTabOption) {
+			newValues = {
+				...newValues,
+				[isNewTabAttrName]: undefined,
+			};
+		}
+
+		setAttributes(newValues);
+
+		setIsDropdownOpen(false);
```

Comment:
Are you sure it will come to this point?

I think that after`setAttributes` code there comes a new state and rerender is coming. Once I had problems with it that's why I'm asking.

Reply:
https://user-images.githubusercontent.com/77000136/126447332-fdf12096-70ce-4150-8aba-18b350fcc9e6.mov

Yup, works as intended. My guess is because it's handling a popover and a seperate state, the re-render of the parent block is not affecting it.